trade-finance-manager-api/src/v1/rest-mappings/facility.js

```diff
@@ -1,22 +1,27 @@
 const mapFacilitySnapshot = require('./mappings/facilities/mapFacilitySnapshot');
 const mapFacilityTfm = require('./mappings/facilities/mapFacilityTfm');
-const mapGefFacility = require('./mappings/gef-facilities/mapGefFacility');
+const mapGefFacilitySnapshot = require('./mappings/gef-facilities/mapGefFacilitySnapshot');
 const isGefFacility = require('./helpers/isGefFacility');
 
+/**
+ * Maps the facility object from the database to a dynamically updating facility object used across TFM.
+ * This facility object updates e.g. when amendments are added in TFM.
+ * @param facility the full facility object from the database
+ * @param dealSnapshot the deal.dealSnapshot object from the database corresponding to the facility
+ * @param dealTfm the deal.dealTfm object from the database corresponding to the facility
+ * @returns mapped facility object for use in TFM
+ */
 const facilityMapper = (facility, dealSnapshot, dealTfm) => {
   const facilityType = facility.facilitySnapshot.type;
+  const isGef = isGefFacility(facilityType);
 
-  if (isGefFacility(facilityType)) {
-    return mapGefFacility(facility, dealSnapshot, dealTfm);
-  }
+  const mappedFacilitySnapshot = isGef ? mapGefFacilitySnapshot(facility, dealSnapshot) : mapFacilitySnapshot(facility, dealSnapshot);
 
-  const result = {
+  return {
```

Comment:
I've refactored to inline returns in a few places (let me know if you'd rather not do this)